README.md

```diff
@@ -51,102 +51,87 @@ function genTokenData(projectNum) {
 let tokenData = genTokenData(123);
 ```
 
-There are two primary ways to use this hash:
+### Safely deriving randomness from the token hash
 
-### 1. Convert the hash to a large integer
-
-The first method is to convert the hash into a large integer and use this to seed a pseudo random number generator \(PRNGs\). You want to consider PRNGs based on arithmetic to ensure your output is deterministic regardless of the JavaScript version being used in a given browser.
-
-```javascript
-let seed = parseInt(tokenData.hash.slice(2, 10), 16)
-```
-
-Here is an example of a PRNG for demonstration purposes - please do your own research about PRNGs and choose one suitable for your needs. Disclaimer: "Anyone who considers arithmetical methods of producing random digits is, of course, in a state of sin." - [John von Neumann](https://en.wikipedia.org/wiki/John_von_Neumann).
-
-#### Xorshift RNGs
-
-George Marsaglia discovered Xorshift RNGs, proposing certain combinations of xorshift operations can result in sufficiently random number generators. For 32-bit inputs - there are many combinations \(a, b, c\) or triplets that could be used, one of which is **\(5,17,13\)**. For more information see the original publication below.
-
-Source: Marsaglia, George. "Xorshift RNGs." Journal of Statistical Software, 8.14 \(2003\): 1 - 6. [10.18637/jss.v008.i14](https://www.jstatsoft.org/article/view/v008i14).  
-Paper: [Creative Commons Attribution 3.0 Unported License](http://creativecommons.org/licenses/by/3.0/).  
-Code: GNU General Public License \(at least one of [version 2](http://www.gnu.org/licenses/gpl-2.0.html) or [version 3](http://www.gnu.org/licenses/gpl-3.0.html) or a [GPL-compatible license](https://gnu.org/licenses/license-list.html#GPLCompatibleLicenses).
-
-Below is a modified shift-register generator derived from the concepts introduced by Marsaglia as well as a C-implementation of the Algorithm "xor" from p. 4 of Marsaglia, "Xorshift RNGs".
+Art Blocks strongly recommends that all artists use an instance of the following Random class to feed all of their project's randomness. 
 
 ```javascript
 class Random {
-  constructor(seed) {
-    this.seed = seed
+  constructor() {
+    this.useA = false;
+    let sfc32 = function (uint128Hex) {
+      let a = parseInt(uint128Hex.substr(0, 8, 16));
+      let b = parseInt(uint128Hex.substr(8, 8, 16));
+      let c = parseInt(uint128Hex.substr(16, 8, 16));
+      let d = parseInt(uint128Hex.substr(24, 8, 16));
+      return function () {
+        a |= 0; b |= 0; c |= 0; d |= 0;
+        var t = (((a + b) | 0) + d) | 0;
+        d = (d + 1) | 0;
+        a = b ^ (b >>> 9);
+        b = (c + (c << 3)) | 0;
+        c = (c << 21) | (c >>> 11);
+        c = (c + t) | 0;
+        return (t >>> 0) / 4294967296;
+      };
+    };
+    this.prngA = new sfc32(tokenData.hash.substr(2, 32));
+    this.prngB = new sfc32(tokenData.hash.substr(34, 32));
```

Comment:
Nit: could you document w/ comments that this takes the first half and second half of the tokenData.hash?